client/src/screens/BallotScreen.tsx

```diff
@@ -1,77 +1,147 @@
-import React, { useContext } from 'react'
-import { useParams } from 'react-router-dom'
+import React, { useContext, useState } from 'react'
+import { useParams, useHistory } from 'react-router-dom'
 import {
   getBallotStyle,
   getContests,
   getPrecinctById,
+  getElectionLocales,
 } from '@votingworks/ballot-encoder'
 import pluralize from 'pluralize'
 
 import { BallotScreenProps } from '../config/types'
 import AppContext from '../contexts/AppContext'
 
-import Button from '../components/Button'
+import Button, { SegmentedButton } from '../components/Button'
 import HandMarkedPaperBallot from '../components/HandMarkedPaperBallot'
 import { Monospace } from '../components/Text'
-import { getBallotFileName } from '../utils/election'
+import {
+  getBallotFileName,
+  getHumanBallotLanguageFormat,
+} from '../utils/election'
 import NavigationScreen from '../components/NavigationScreen'
 import HorizontalRule from '../components/HorizontalRule'
+import { DEFAULT_LOCALE } from '../config/globals'
+import { routerPaths } from '../components/ElectionManager'
 
 const BallotScreen = () => {
-  const { precinctId, ballotStyleId } = useParams<BallotScreenProps>()
+  const history = useHistory()
+  const {
+    precinctId,
+    ballotStyleId,
+    localeCode: currentLocaleCode,
+  } = useParams<BallotScreenProps>()
   const { election: e, electionHash } = useContext(AppContext)
   const election = e!
+  const availableLocaleCodes = getElectionLocales(election, DEFAULT_LOCALE)
+
   const precinctName = getPrecinctById({ election, precinctId })?.name
   const ballotStyle = getBallotStyle({ ballotStyleId, election })!
-
   const ballotContests = getContests({ ballotStyle, election })
 
+  const [isLiveMode, setIsLiveMode] = useState(true)
+  const toggleLiveMode = () => setIsLiveMode((m) => !m)
+  const changeLocale = (localeCode: string) =>
+    history.replace(
```

Comment:
do you mean to do a `replace` here and thus clear the previous history entry?

Reply:
Yes, I do mean to replace. I would like there to be a URL for these ballots, however there is no need to have this URL be captured when clicking buttons such that the back button works as expected. But… now that I'm thinking about it, this app will be used inside kiosk browser where there is no back button and because this used on a toggle UI element… this really won't matter either way. I'm gonna leave it as is as having a url for this does provide utility in dev env.